### PR TITLE
feat(lightspeed-backend)!: migrate to the new backend and remove deprecations

### PR DESCRIPTION
**Pull request by dzemanov**
## General
- removes legacy backend system
- removes deprecations
## Fixes
[RHIDP-4229](https://issues.redhat.com/browse/RHIDP-4229)

### CODE REVIEW

File: plugins/lightspeed-backend/src/service/router.test.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
-import { getVoidLogger } from '@backstage/backend-common';
-// import { mockCredentials, mockServices } from '@backstage/backend-test-utils';
+import { mockServices } from '@backstage/backend-test-utils';
 import { MockConfigApi } from '@backstage/test-utils';
```
Comment (schultzp2020) [inline plugins/lightspeed-backend/src/service/router.test.ts:2]: We should use mock services [here](https://github.com/janus-idp/backstage-plugins/blob/main/plugins/lightspeed-backend/src/service/router.test.ts#L58) instead.

File: plugins/lightspeed-backend/src/service/types.ts [hunk 2]
```diff
@@ -2,10 +2,10 @@ import {
   // DiscoveryService,
```
Comment (schultzp2020) [inline plugins/lightspeed-backend/src/service/types.ts:2]: Is there a reason why the types are commented out?

Comment (dzemanov) [reply plugins/lightspeed-backend/src/service/types.ts:2]: I guess @yangcao77 added these for future reference? https://github.com/janus-idp/backstage-plugins/blob/main/plugins/lightspeed-backend/src/plugin.ts#L22-L24
I can remove them if @yangcao77 is ok with it.

Comment (yangcao77) [reply plugins/lightspeed-backend/src/service/types.ts:2]: I will  those for user authentication, but feel free to remove it if you would like to.

**Review by yangcao77 — APPROVED** [review]
lgtm, thanks for the change to migrate to new backend

File: plugins/lightspeed-backend/src/service/router.test.ts [hunk 3]
```diff
@@ -1,6 +1,4 @@
-import { getVoidLogger } from '@backstage/backend-common';
-// import { mockCredentials, mockServices } from '@backstage/backend-test-utils';
-import { MockConfigApi } from '@backstage/test-utils';
+import { mockServices } from '@backstage/backend-test-utils';
```
Comment (schultzp2020) [inline plugins/lightspeed-backend/src/service/router.test.ts:1]: This test file should be refactored to follow the structure of the ocm `router.test.ts` file.

File: plugins/lightspeed-backend/src/service/types.ts [hunk 4]
```diff
@@ -2,10 +2,10 @@ import {
...
   // HttpAuthService,
   // UserInfoService,
-  LoggerService,
+  type LoggerService,
```
Comment (schultzp2020) [inline plugins/lightspeed-backend/src/service/types.ts:5]:
```suggestion
  LoggerService,
```
Move `type` to the top level and remove the comments.

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

File: plugins/lightspeed-backend/src/service/router.test.ts [hunk 5]
```diff
@@ -85,35 +46,50 @@ jest.mock('http-proxy-middleware', () => ({
...
-  });
+  afterAll(() => server.close());
 
-  beforeEach(() => {
+  afterEach(() => {
     jest.clearAllMocks();
+    server.resetHandlers();
   });
 
+  async function startBackendServer(config?: any) {
```
Comment (schultzp2020) [inline plugins/lightspeed-backend/src/service/router.test.ts:75]:
```suggestion
  async function startBackendServer(config?: Record<PropertyKey, unknown>) {
```

**Review by schultzp2020 — CHANGES_REQUESTED** [review]

**Review by schultzp2020 — APPROVED** [review]
/approve
/lgtm

**Pull request merged**